lib/twingly/public_suffix_list.rb

```diff
@@ -0,0 +1,35 @@
+require "public_suffix"
+
+module Twingly
+  class PublicSuffixList
+    ACE_PREFIX = /\Axn\-\-/i.freeze
+
+    private_constant :ACE_PREFIX
+
+    # Extend the PSL with ASCII form of all internationalized domain names
+    def self.with_punycoded_names
+      list_data = File.read(PublicSuffix::List::DEFAULT_LIST_PATH)
+      list = PublicSuffix::List.parse(list_data, private_domains: false)
+
+      punycoded_names(list).each do |punycoded_name|
+        new_rule = PublicSuffix::Rule.factory(punycoded_name)
+        list.add(new_rule, reindex: false)
+      end
+
+      list.reindex!
+
+      list
+    end
+
+    private_class_method def self.punycoded_names(list)
```

Comment:
An alternative could be:

```Ruby
private_class_method \
def self.punycoded_names(list)
  # ..
end
```

Not sure which I prefer though (probably one with newline).